receiver/haproxyreceiver/scraper.go

```diff
@@ -28,30 +32,60 @@ var (
 )
 
 type scraper struct {
-	endpoint string
-	logger   *zap.Logger
-	mb       *metadata.MetricsBuilder
+	cfg               *Config
+	httpClient        *http.Client
+	logger            *zap.Logger
+	mb                *metadata.MetricsBuilder
+	telemetrySettings component.TelemetrySettings
 }
 
 func (s *scraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
-	var d net.Dialer
-	c, err := d.DialContext(ctx, "unix", s.endpoint)
-	if err != nil {
-		return pmetric.NewMetrics(), err
-	}
-	defer func(c net.Conn) {
-		_ = c.Close()
-	}(c)
-	records, err := s.readStats(c)
-	if err != nil {
-		return pmetric.NewMetrics(), err
+
+	var records []map[string]string
+	if u, notURLerr := url.Parse(s.cfg.Endpoint); notURLerr == nil && (u.Scheme == "http" || u.Scheme == "https") {
+
+		resp, err := s.httpClient.Get(s.cfg.Endpoint + ";csv")
+		if err != nil {
+			return pmetric.NewMetrics(), err
+		}
+		defer resp.Body.Close()
+		buf, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return pmetric.NewMetrics(), err
+		}
+		records, err = s.readStats(buf)
+		if err != nil {
+			return pmetric.NewMetrics(), err
+		}
+	} else {
+		var d net.Dialer
+		c, err := d.DialContext(ctx, "unix", s.cfg.Endpoint)
+		if err != nil {
+			return pmetric.NewMetrics(), err
+		}
+		defer func(c net.Conn) {
```

Comment:
I wonder what's the different between this and `defer c.Close()`, is this func is used on purpose?

Reply:
this is just to make explicit that we're not going to care for the error returned by Close() by assigning it to `_`, which removes a warning from IDEs.